lib/eddystone-beacon-scanner.js

```diff
@@ -24,47 +32,188 @@ EddystoneBeaconScanner.prototype.startScanning = function(allowDuplicates) {
       noble.once('stateChange', startScanningOnPowerOn);
     }
   };
+  
+  var didLoseBeacon = setInterval(this.didLoseBeacon.bind(this), EXIT_GRACE_PERIOD / 2);
 
   startScanningOnPowerOn();
 };
 
 EddystoneBeaconScanner.prototype.stopScanning = function() {
   debug('stopScanning');
+  
   noble.stopScanning();
+  clearInterval(didLoseBeacon);
 };
 
-EddystoneBeaconScanner.prototype.onDiscover = function(peripheral) {
-  debug('onDiscover: %s', peripheral);
+EddystoneBeaconScanner.prototype.didFindBeacon = function(peripheral) {
+  debug('didFindBeacon: %s', peripheral);
+
+  if (this.isEddyStoneBeacon(peripheral) === true) {
 
-  if (this.isUrlBeacon(peripheral)) {
-    var urlBeacon = this.parseUrlBeacon(peripheral);
+    this.cacheBeacon(peripheral);
 
-    debug('onDiscover: urlBeacon = %s', JSON.stringify(urlBeacon));
-    this.emit('discover', urlBeacon);
+    var beaconInfo = this.parseBeacon(peripheral);
+
+    debug('didFindBeacon: Beacon = %s', JSON.stringify(beaconInfo));
+    this.emit('didFindBeacon', beaconInfo);
   }
 };
 
-EddystoneBeaconScanner.prototype.isUrlBeacon = function(peripheral) {
+EddystoneBeaconScanner.prototype.didLoseBeacon = function() {
+  for (var id in inRange) {
+    if (inRange[id].lastSeen < (Date.now() - EXIT_GRACE_PERIOD)) {
+      var peripheral = inRange[id].peripheral;
+      var lostBeacon = this.parseBeacon(peripheral);
```

Comment:
I find that when a beacon is lost the peripheral doesnt have any service data in it and this crashes. Is there actually a case where we have service data on an exiting beacon? Im using node-eddystone-beacon uid with-tlm example.